src/servers/src/mysql/handler.rs

```diff
@@ -102,28 +108,44 @@ impl<W: AsyncWrite + Send + Sync + Unpin> AsyncMysqlShim<W> for MysqlInstanceShi
 
     async fn authenticate(
         &self,
-        _auth_plugin: &str,
+        auth_plugin: &str,
         username: &[u8],
         salt: &[u8],
         auth_data: &[u8],
     ) -> bool {
         // if not specified then **root** will be used
         let username = String::from_utf8_lossy(username);
         let client_addr = self.client_addr.clone();
-        let auth_method = match auth_data.len() {
-            0 => AuthMethod::None,
-            _ => AuthMethod::Password {
-                hash_method: DoubleSha1,
-                hashed_value: auth_data.to_vec(),
-                salt: salt.to_vec(),
-            },
-        };
+
+        if let Some(user_provider) = &self.user_provider {
+            let user_id = Identity::UserId(&username, Some(&client_addr));
+
+            match user_provider.user_info(user_id.clone()).await {
+                Ok(userinfo) => {
+                    let pwd = match auth_plugin {
+                        "mysql_native_password" => Password::MysqlNativePwd(auth_data, salt),
+                        other => {
+                            error!("Unsupported mysql auth plugin: {}", other);
+                            return false;
+                        }
+                    };
+                    if !userinfo.auth_method().auth(pwd) {
+                        return false;
+                    }
+                }
+                Err(e) => {
+                    error!("Failed to auth, err: {:?}", e);
+                    return false;
+                }
+            }
+        }
 
         return match CtxBuilder::new()
-            .client_addr(Some(client_addr))
-            .set_channel(Some(MYSQL))
-            .set_username(Some(username.to_string()))
-            .set_auth_method(Some(auth_method))
+            .client_addr(client_addr)
+            .set_channel(MYSQL)
+            .set_user_info(UserInfo {
+                username: username.to_string(),
```

Comment:
if `UserProvider` is not provided and username is not specified while receiving client request, the default username is `root`, consider following
1. if `UserProvider` is not provided, return `UserInfo::default` instead of input username
2. change username to `greptime` rather than use input(or default) username

I would prefer the first one